pkg/eth/api.go

```diff
@@ -524,12 +524,18 @@ func (pea *PublicEthAPI) localGetTransactionReceipt(ctx context.Context, hash co
 		"logsBloom":         receipt.Bloom,
 	}
 
-	// Assign receipt status or post state.
-	if len(receipt.PostState) > 0 {
+	if blockNumber <= pea.B.Config.ChainConfig.ByzantiumBlock.Uint64() {
+		if receipt.GasUsed > tx.Gas() {
+			fields["status"] = hexutil.Uint(types.ReceiptStatusFailed)
+		} else {
+			fields["status"] = hexutil.Uint(types.ReceiptStatusSuccessful)
+		}
+	} else if len(receipt.PostState) > 0 {
```

Comment:
We still want to fill in the `fields["root"]` if receipt.PostState is present, whether or not we can also assume the status above.